File: cases/taint/c/branch-join-positive/branch_join_positive.c
int dfb_source(void) { // DFB-SOURCE: branch-join-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: branch-join-sink

void run(int overwrite) {
    int value = dfb_source();
    if (overwrite) {
        value = 0;
    }
    // DFB-WITNESS: branch-join-value
    dfb_sink(value);
}

File: cases/taint/kotlin/branch-join-positive/BranchJoinPositive.kt
package dataflowbench

object BranchJoinPositive {
    fun dfb_source(): Int { // DFB-SOURCE: branch-join-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: branch-join-sink

    fun run(overwrite: Boolean) {
        var value = dfb_source()
        if (overwrite) {
            value = 0
        }
        // DFB-WITNESS: branch-join-value
        dfb_sink(value)
    }
}
